eva-accession-release/src/test/java/uk/ac/ebi/eva/accession/release/io/MergedVariantMongoReaderTest.java

```diff
@@ -0,0 +1,219 @@
+                allVariants.put(getStringId(variant), variant);
+            }
+        }
+        reader.close();
+        return allVariants;
+    }
+
+    private String getStringId(Variant variant) {
+        return (variant.getChromosome() + "_" + variant.getStart() + "_" + variant.getReference() + "_"
+                + variant.getAlternate()).toUpperCase();
+    }
+
+    @Test
+    public void checkMergedInto() throws Exception {
+        Map<String, Variant> variants = readIntoMap();
+        assertEquals(3, variants.size());
+
+        assertTrue(variants.get(ID_1_A)
+                           .getSourceEntries()
+                           .stream()
+                           .allMatch(e -> ID_1_MERGED_INTO.equals(e.getAttribute(MERGED_INTO_KEY))));
+
+        assertTrue(variants.get(ID_1_T)
+                           .getSourceEntries()
+                           .stream()
+                           .allMatch(e -> ID_1_MERGED_INTO.equals(e.getAttribute(MERGED_INTO_KEY))));
+
+        assertTrue(variants.get(ID_2)
+                           .getSourceEntries()
+                           .stream()
+                           .allMatch(e -> ID_2_MERGED_INTO.equals(e.getAttribute(MERGED_INTO_KEY))));
+    }
+
+    @Test
+    public void checkAlleles() throws Exception {
+        Map<String, Variant> variants = readIntoMap();
+        assertEquals(3, variants.size());
+
+        assertEquals("G", variants.get(ID_1_A).getReference());
+        assertEquals("A", variants.get(ID_1_A).getAlternate());
+
+        assertEquals("G", variants.get(ID_1_T).getReference());
+        assertEquals("T", variants.get(ID_1_T).getAlternate());
+
+        assertEquals("T", variants.get(ID_2).getReference());
+        assertEquals("G", variants.get(ID_2).getAlternate());
+    }
+
+    @Test
+    public void includeValidatedFlag() throws Exception {
+        assertFlagEqualsInAllVariants(CLUSTERED_VARIANT_VALIDATED_KEY, false);
+        assertFlagEqualsInAllVariants(SUBMITTED_VARIANT_VALIDATED_KEY, false);
+    }
+
+    private void assertFlagEqualsInAllVariants(String key, boolean value) throws Exception {
+        Map<String, Variant> variants = readIntoMap();
+        assertNotEquals(0, variants.size());
+        assertTrue(variants.values().stream()
+                           .flatMap(v -> v.getSourceEntries().stream())
+                           .map(se -> se.getAttribute(key))
+                           .map(Boolean::new)
+                           .allMatch(v -> v.equals(value)));
+    }
+
+    @Test
+    public void includeAssemblyMatchFlag() throws Exception {
+        assertFlagEqualsInAllVariants(ASSEMBLY_MATCH_KEY, true);
+    }
+
+    @Test
+    public void includeAllelesMatchFlag() throws Exception {
+        assertFlagEqualsInAllVariants(ALLELES_MATCH_KEY, true);
+    }
+
+    @Test
+    public void includeEvidenceFlag() throws Exception {
+        assertFlagEqualsInAllVariants(SUPPORTED_BY_EVIDENCE_KEY, true);
+    }
+
+}
```

Comment:
I could add more tests, but I'm not sure we need to put them because it's already tested in the parent class, here we test that the "mergedInto" attribute was picked up correctly, and also the alleles and flags from the joined collection